esp-hal/Cargo.toml

```diff
@@ -183,3 +160,48 @@ defmt = [
     "embedded-io/defmt-03",
     "embedded-io-async?/defmt-03",
 ]
+## Implement the traits defined in the `1.0.0` releases of `embedded-hal` and
+## `embedded-hal-nb` for the relevant peripherals.
+eh1 = ["embedded-hal-1", "embedded-hal-nb", "embedded-can"]
+## Implement the traits defined in `embedded-io` for certain peripherals.
+embedded-io = ["dep:embedded-io"]
+## Implement the `ufmt_write::uWrite` trait for certain peripherals.
+ufmt = ["dep:ufmt-write"]
+
+#! ### Embassy Feature Flags
+## Enable support for `embassy`, a modern asynchronous embedded framework.
+embassy = ["embassy-time-driver", "procmacros/embassy"]
+## Use the interrupt-mode embassy executor.
+embassy-executor-interrupt = ["embassy", "embassy-executor"]
+## Use the thread-mode embassy executor.
+embassy-executor-thread = ["embassy", "embassy-executor"]
+## Uses hardware timers as alarms for the executors. Using this feature
+## limits the number of executors to the number of hardware alarms provided
+## by the time driver.
+embassy-integrated-timers = ["embassy-executor?/integrated-timers"]
+## Enable the embassy time driver using the `SYSTIMER` peripheral. The
+## `SYSTIMER` peripheral has three alarams available for use.
+embassy-time-systick = []
+## Enable the embassy time driver using the `TIMG0` peripheral. The `TIMG0`
+## peripheral has two alarms available for use.
+embassy-time-timg0 = []
+
+#! ### PSRAM Feature Flags
+## Use externally connected PSRAM (2MB).
+psram-2m = []
+## Use externally connected PSRAM (4MB).
+psram-4m = []
+## Use externally connected PSRAM (8MB).
+psram-8m = []
+## PSRAM 80Mhz frequency support
+psram-80mhz = []
+
+#! ### Octal RAM Feature Flags
+## Use externally connected Octal RAM (2MB).
+opsram-2m  = []
+## Use externally connected Octal RAM (4MB).
+opsram-4m  = []
+## Use externally connected Octal RAM (8MB).
+opsram-8m  = []
+## Use externally connected Octal RAM (16MB).
+opsram-16m = []
```

Comment:
any of the psram features need to activate a feature "psram"

Reply:
we need that only for S3 and only because of https://github.com/jessebraham/esp-hal/blob/7763036e47e5fb695b4e39197c033bc77779692a/esp-hal/build.rs#L439C44-L443

We could also list all the psram features there - not sure what is less annoying

Reply:
Or ... we just always reserve the bigger cache